src/DDTrace/Integrations/Curl/CurlIntegration.php

```diff
@@ -25,6 +25,46 @@ function addSpanDataTagFromCurlInfo($span, &$info, $tagName, $curlInfoOpt)
     }
 }
 
+ /**
+  * Determines if a given status code should be considered an error
+  * based on the DD_TRACE_HTTP_CLIENT_ERROR_STATUSES configuration.
+  *
+  * @param int $statusCode The HTTP status code to check
+  * @return bool Whether the status code should be considered an error
+  */
+private static function isClientError($statusCode) {
+    // Get configured status codes from environment
+    $errorStatusCodes = \dd_trace_env_config("DD_TRACE_HTTP_CLIENT_ERROR_STATUSES");
+
+    if (!empty($errorStatusCodes)) {
+        // Custom configuration exists, use it
+        $codesList = explode(',', $errorStatusCodes);
+
+        foreach ($codesList as $item) {
+            $item = trim($item);
+
+            if (strpos($item, '-') !== false) {
+                // Range ("400-499")
+                list($start, $end) = explode('-', $item);
+                if ($statusCode >= (int)$start && $statusCode <= (int)$end) {
+                    return true;
+                }
+            } else {
+                // Single code ("404")
+                if ($statusCode == (int)$item) {
+                    return true;
+                }
+            }
+        }
+
+        // The status code isn't in any defined error range
+        return false;
+    } else {
+        // Default behavior
+        return ($statusCode >= 400);
+    }
+}
```

Comment:
Can we add this common code to DDTrace\Integrations\HttpClientIntegrationHelper please? Thanks!